src/plugins/data/public/ui/dataset_selector/configurator.tsx

```diff
@@ -52,25 +53,57 @@ export const Configurator = ({
       defaultMessage: "I don't want to use the time filter",
     }
   );
-  const [language, setLanguage] = useState<string>(() => {
+  const [selectedLanguage, setSelectedLanguage] = useState<string>('');
+  const indexedViewsService = type?.indexedViewsService;
+  const [selectedIndexedView, setSelectedIndexedView] = useState<string | undefined>();
+  const [indexedViews, setIndexedViews] = useState<IndexedView[]>([]);
+  const [isLoadingIndexedViews, setIsLoadingIndexedViews] = useState(false);
+
+  useEffect(() => {
+    setLanguages(type?.supportedLanguages(dataset) || []);
+  }, [dataset, type]);
+
+  useEffect(() => {
     const currentLanguage = queryString.getQuery().language;
     if (languages.includes(currentLanguage)) {
-      return currentLanguage;
+      setSelectedLanguage(currentLanguage);
+      return;
     }
-    return languages[0];
-  });
+    setSelectedLanguage(languages[0]);
```

Comment:
I noticed that the default value for `selectedLanguage` was `''`. This line could potentially set it to `undefined`. Is the logic capable of handling that?

Reply:
Do we even do any validation on these languages? What would happen if `selectedLanguage = 'invalid'`? It would be great if we could add some comments (if none exist already) about the treatment of unset, empty, and other invalid values.

Reply:
This PR is not changing the existing behavior for languages and only making it reactive so that we can decide whether to show the indexed views. I would suggest we add validations with proper plan for handling the invalid values